x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -125,9 +152,14 @@ export function toggleLayerVisible(layerId) {
 }
 
 export function setSelectedLayer(layerId) {
-  return {
-    type: SET_SELECTED_LAYER,
-    selectedLayerId: layerId
+  return  dispatch => {
```

Comment:
Maybe this needs another step that checks for unsaved state in existing layer and if there is dispatches another event that would is display a popup to the user saying "You have unsaved changes on the layer do you want to continue and loose them"

Reply:
Agreed. will add modal-flow